packages/api/src/elastic/index.ts

```diff
@@ -0,0 +1,232 @@
+/* eslint-disable @typescript-eslint/no-unsafe-member-access */
+/* eslint-disable @typescript-eslint/no-unsafe-assignment */
+import { env } from '../env'
+import { Client } from '@elastic/elasticsearch'
+import { Label, PageType } from '../generated/graphql'
+
+export type Page = {
+  id?: string
+  userId: string
+  title: string
+  author?: string
+  description?: string
+  content: string
+  url?: string
+  hash: string
+  uploadFileId?: string
+  image?: string
+  pageType: PageType
+  originalHtml?: string
+  slug: string
+  labels?: Label[]
+  readingProgress: number
+  readingProgressAnchorIndex: number
+  createdAt: Date
+  updatedAt: Date
+  publishedAt?: Date
+  savedAt?: Date
+  sharedAt?: Date
+  archivedAt?: Date
+}
```

Comment:
@jacksonh added elastic index (not including searching queries), schema and util methods here